File: Web/RMA-Web-Shell/projects/fincare/src/app/billing-manager/models/InvoicePaymentAllocation.ts
import { InvoiceStatusEnum } from 'projects/shared-models-lib/src/lib/enums/invoice-status-enum';

export class InvoicePaymentAllocation {
  invoiceId: number;
  policyNumber: string;
  collectionDate: Date;
  invoiceDate: Date;
  totalInvoiceAmount: number;
  invoiceStatus: InvoiceStatusEnum;
  invoiceNumber: string;
  amountOutstanding: number;
  amountAllocated: number;
  displayName: string;
  isClaimRecoveryInvoice: boolean;
}

File: Service/Billing/RMA.Service.Billing.Contracts/Entities/InvoicePaymentAllocation.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.Billing.Contracts.Entities
{
    public class InvoicePaymentAllocation
    {
        public int InvoiceId { get; set; }
        public string PolicyNumber { get; set; }
        public System.DateTime CollectionDate { get; set; }
        public System.DateTime InvoiceDate { get; set; }
        public decimal TotalInvoiceAmount { get; set; }
        public InvoiceStatusEnum InvoiceStatus { get; set; }
        public string InvoiceNumber { get; set; }
        public string DisplayName { get; set; }
        public decimal AmountOutstanding { get; set; }
        public bool IsClaimRecoveryInvoice { get; set; }
    }
}
